packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -245,11 +295,48 @@ export default {
     },
     filteredAppMap() {
       const { appMap } = this.$store.state;
-      const events = appMap.rootEvents().reduce((callTree, rootEvent) => {
+      let rootEvents = appMap.rootEvents();
+
+      if (this.filters.limitRootEvents.on) {
+        rootEvents = rootEvents.filter((e) => e.httpServerRequest);
+      }
+
+      let events = rootEvents.reduce((callTree, rootEvent) => {
         rootEvent.traverse((e) => callTree.push(e));
         return callTree;
       }, []);
 
+      events = events.filter(
+        (e) => this.filters.unlabeledEvents.on || e.labels.size > 0
```

Comment:
```js
this.filters.unlabeledEvents.on ||
e.labels.size > 0 ||
e.codeObject.type !== CodeObjectType.FUNCTION
```

This filter should only apply to functions.

Reply:
Added new condition